### DISCUSSION TITLE
Moose installation without sudo permissions _glibc older versions

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#28513] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello
I recently tried to upgrade moose on my university server (without sudo permissions).
1) Creating conda environment using `conda create -n moose moose-dev=2024.08.26 mpich`
gives me error to update to update __glibc to 2.28 or higher as follow:
```
Channels:
 - https://conda.software.inl.gov/public
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides __glibc >=2.28,<3.0.a0 needed by c-ares-1.33.0-ha66036c_0

Could not solve for environment specs
The following package could not be installed
└─ moose-dev 2024.08.26**  is not installable because it requires
   └─ moose-peacock 2024.08.12.* , which requires
      └─ moose-mpi 2024.08.12.* , which requires
         └─ c-ares 1.33.0 ha66036c_0, which requires
            └─ __glibc >=2.28,<3.0.a0 , which is missing on the system.
```
Upgrading __glibc requires me to update a number of packages that is a tedious task as most of the gnu softwares installed in /bin at server is outdated.

Instead of doing this:
I created conda environment using
`conda create -n moose moose mpich`
So, I get a warning during moose installation as:
`WARNING: (CondaVersionMismatch(...), "Conda package 'moose-dev' is currently at version '2024.07.19' and the required version is '2024.08.26'.\nThe correct version can be installed via:\n    conda install moose-dev=2024.08.26")`

I have ignored this warning for time being as I dont want to upgrade the __glibc right now.

But doing this makes me not able to run the mpiexec command for utilizing parallel processing. The error is as follow:
```

SLURM_CLUSTER_NAME = param-shakti
SLURM_JOB_ACCOUNT = sreerajes
SLURM_JOB_ID = 1537885
SLURM_JOB_NAME = tiny
SLURM_JOB_NODELIST = cn[039-040]
SLURM_JOB_USER = 19es92r03
SLURM_JOB_UID = 6905
SLURM_JOB_PARTITION = shared
SLURM_TASK_PID = 49164
SLURM_SUBMIT_DIR = /scratch/19es92r03/O4_progress
SLURM_CPUS_ON_NODE = 35
SLURM_NTASKS = 36
SLURM_TASK_PID = 49164
==========================================

^[[33m
*** Warning, This code is deprecated and will be removed in future versions:
/scratch/19es92r03/O4_progress/input_original_updated.i:350: (Outputs/other/interval):
'interval' has been deprecated and will be removed on 02/01/2025. Please use 'time_step_interval' instead.
^[[39m
AutoDiff exception: Unknown serialization file version (this message will only be shown once per process)

^[[31m
*** ERROR ***
Failed to take order 1 derivative in material Mobility_coefficient^[[39m

Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

I am in a hurry, so should I downgrade the moose version to conda compatible, or is there any other way out? Also is this parallel processing error solely due to conda and moose mismatch or some other issue ? 
The older version of moose worked fine on multiple processing on server.

Thanks

**Comment by milljm** [comment, marked as answer]
The only thing you can do in this case, is building the entire stack on your own, without the use of our Conda packages. Indeed, when running on HPC Clusters, it is more desirable _not_ using Conda packages.

So in essence, you'd be doing the right thing by utilizing your cluster's native module(s) system.
